tools/src/main/scala/scala/scalanative/codegen/Lower.scala

```diff
@@ -269,92 +270,97 @@ object Lower {
       }
     }
 
-    def genLet(buf: Buffer, n: Local, op: Op): Unit = op.resty match {
-      case Type.Unit =>
-        genOp(buf, fresh(), op)
-        buf.let(n, Op.Copy(unit), unwind)
-      case Type.Nothing =>
-        genOp(buf, fresh(), op)
-        genUnreachable(buf)
-        buf.label(fresh(), Seq(Val.Local(n, op.resty)))
-      case _ =>
-        genOp(buf, n, op)
-    }
+    def genLet(buf: Buffer, n: Local, op: Op)(implicit pos: Position): Unit =
+      op.resty match {
+        case Type.Unit =>
+          genOp(buf, fresh(), op)
+          buf.let(n, Op.Copy(unit), unwind)(Position.generated)
```

Comment:
In this case here as well, never use a generated position for code that is the product of transforming a tree. You should carry the position that you receive from above.